### PR TITLE
Add preprocess_entity to replace preprocess_body

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: pulpcore/cli/common/generic.py [hunk 1]
```diff
@@ -852,7 +849,7 @@ def callback(pulp_ctx: PulpContext, entity_ctx: PulpEntityContext, **kwargs: Any
         """
         Create a {entity}.
         """
-        body: EntityDefinition = entity_ctx.preprocess_body(kwargs)
+        body: EntityDefinition = entity_ctx.preprocess_entity(kwargs, partial=False)
```
Comment (mdellweg) [inline pulpcore/cli/common/generic.py:852]: I would like to move the call for `preprocess_entity` into the `create` and `update` calls, but it's not idempotent, and we may run into trouble with external plugins. Suggestions?

Comment (gerrod3) [reply pulpcore/cli/common/generic.py:852]: - Cache the results of `preprocess_entity`
- Add a field to result of `preprocess_entity` that `create` and `update` would remove and know not to run it a second time
- Change the function signature of `create` and `update` to take in an unprocessed body
- In `create` & `update` run `preprocess_entity` on a copy of `body` and then **magically** chose the correct one to use.

Comment (mdellweg) [reply pulpcore/cli/common/generic.py]: I went with a slight modification of your second suggestion: A subclass of dict.

File: pulpcore/cli/common/context.py [hunk 2]
```diff
@@ -60,6 +64,21 @@ def default(self, obj: Any) -> Any:
             return super().default(obj)
 
 
+def _preprocess_value(key: str, value: Any) -> Any:
+    if isinstance(value, PulpEntityContext):
+        return value.pulp_href
+    return value
+
+
+def preprocess_body(body: EntityDefinition) -> EntityDefinition:
```
Comment (mdellweg) [inline pulpcore/cli/common/context.py:73]: We now preprocess body _and_ parameters with this. So maybe it is not the best name for the function.

I'm going to call it `preprocess_payload`.

**Review by gerrod3 — APPROVED** [review]
Nice, I like the option you went with.

**Pull request merged**